commons/all/src/main/java/org/infinispan/commons/util/Util.java

```diff
@@ -633,12 +633,12 @@ public static String hexDump(ByteBuffer buffer) {
    }
 
    public static String hexDump(byte[] buffer, int actualLength) {
-      int dumpLength = Math.min(buffer.length, HEX_DUMP_LIMIT);
+      int dumpLength = Math.min(Math.min(buffer.length, HEX_DUMP_LIMIT), actualLength);
       StringBuilder sb = new StringBuilder(dumpLength * 2 + 30);
       for (int i = 0; i < dumpLength; ++i) {
          addHexByte(sb, buffer[i]);
       }
-      if (dumpLength < actualLength) {
+      if (dumpLength < HEX_DUMP_LIMIT) {
```

Comment:
I don't think this is correct.
`dumpLength` must be less or equals to `HEX_DUMP_LIMIT`. 
The original condition is correct IMO.